src/Calculator.ts

```diff
@@ -116,20 +118,109 @@ export function calculate(
  * @param measureBundle Bundle with a MeasureResource and all necessary data for execution.
  * @param patientBundles List of bundles of patients to be executed.
  * @param options Options for calculation.
- * @returns MeasureReport resource for each patient.
+ * @returns MeasureReport resource for each patient according to standard https://www.hl7.org/fhir/measurereport.html
  */
 export function calculateMeasureReports(
   measureBundle: R4.IBundle,
   patientBundles: R4.IBundle[],
   options: CalculationOptions
 ): R4.IMeasureReport[] {
-  return [
-    {
-      resourceType: 'MeasureReport',
-      measure: 'Measure/ImplementMe',
-      period: {}
-    }
-  ];
+  // options should be updated by this call if measurementPeriod wasn't initially passed in
+  const results: ExecutionResult[] = calculate(measureBundle, patientBundles, options);
+  const reports: R4.IMeasureReport[] = [];
+  results.forEach(function (result) {
+    const report: R4.IMeasureReport = <any>{};
```

Comment:
```suggestion
    const report = <R4.IMeasureReport>{};
```

I'm assuming the reasoning for the `any` cast here is that this interface requires specific information and it won't let you initialize it to empty object. We then later fill out the proper information in the report in subsequent lines. The "proper" way to do this in typescript is to omit a type from the left side and use a type assertion for the desired type on the right side. This essentially tells the compiler: "dang trust me, this is a measure report"

There are a couple more instances of this that I'll add suggestions for as well